docs/troubleshooting/agent.md

```diff
@@ -0,0 +1,109 @@
+s5248-05       server-leaf   leaf-1        DellEMC-S5248f-P-25G-DPB   broadcom   16s         85s       38         38         v0.84.3   4.5.0-Enterprise_Base   85s       38         true   true                 18d
+ds3000-01      server-leaf   leaf-5        DS3000                     broadcom   10s         23m       37         37         v0.84.3   4.5.0-Enterprise_Base   23m       37                                     18d
+```
+
+## Heart Beats
+
+Heart beats are messages between the controller and the switches. The purpose
+of sending these messages is to confirm that the switches are reachable and
+responding to the commands of the controller. To check the status of the agent:
+
+```bash
+systemctl status hedgehog-agent.service
+```
+
+### Diagnosing Long or Missing Heart beats
+
+If the duration between heart beats is greater than 1 minute, one of several
+things could be happening:
+
+* The connection between the switch and the control node is down or misconfigured.
+* The agent isn't running on the switches
+* The agent is stuck trying to apply a configuration
+* The time is not synchronized between the controller and the switches 
+
+#### Switch and Controller Connection
+
+The control node serves DHCP to the switches as part of normal operation and
+replies to ONIE requests for NOS install. The controller expects layer 2
+reachability and to be the only DHCP server in the LAN. This architecture
+can take many forms but it is most often a dedicated switch or VLAN.
+Ensure that the link on the control node is in the `up` state. Often problems
+in this area are related to the configuration of a private VLAN, or cabling.
+
+#### Starting and Stopping the Agent
+
+The switch agent is a service that is controlled via `systemctl` commands.
+The agent is started automatically at switch bootup time, to stop the agent:
+```bash
+sudo systemctl stop hedgehog-agent.service
+```
+similarly to start the agent:
+```bash
+sudo systemctl start hedgehog-agent.service
+```
+
+Stopping the agent should only be done as part of debugging, because the agent will
+reapply the configuration on a regular interval during normal operation.
+
+#### Restoring Agent State
+
+In rare circumstances, the agent can become stuck trying to apply a
+configuration. Before attempting these steps reach out to Hedgehog as this is
+likely a bug. If an agent is stuck, the configuration can be
+generated on the control node using `kubectl`, then manually moved to the
+switch. The example below uses the name of a switch for illustrative purposes.
+
+1. on control node: `kubectl get -o yaml agents/switch-name > agent-config.yaml`
+1. on control node: `scp agent-config.yaml admin@leaf-04:/tmp/agent-config.yaml`
+1. on control node: `ssh admin@leaf-04`
+1. on switch: `sudo mv /tmp/agent-config.yaml /etc/sonic/hedgehog/agent-config.yaml`
+
```

Comment:
Probably explaining that agent will restart and re-read it would be good. Additionally, if it's not helping then switch reboot may be required